software/base/src/main/java/brooklyn/entity/basic/SoftwareProcessImpl.java

```diff
@@ -456,6 +456,9 @@ protected final void callStartHooks() {}
                 Object value = getConfig(k);
                 if (value instanceof Integer){
                     ports.add((Integer)value);
+                } else if (PortRange.class.isAssignableFrom(value.getClass())) {
```

Comment:
ok. But still better to pick it up on all keys that are a PortRange, not just those ending .port, right?